src/services/messages-processor/service.ts

```diff
@@ -269,11 +284,35 @@ export const makeMessagesProcessor = ({
     }
   }
 
-  const readFolder = async (uri: string): Promise<string[]> => {
+  const readFolder = async (uri: string): Promise<MessageFile[]> => {
     if (uri.startsWith('s3://')) return s3Service.read(uri)
     if (uri.startsWith('gs://')) return gsService.read(uri)
     return localFileReader.readFilesFromFolder(uri)
   }
```

Comment:
Consider implementing a single interface for these services

Reply:
Hm... Because of services structure it's hard to add one interface here. If classes were used it would be easier.